plugins/modules/events_and_notifications_workflow_manager.py

```diff
@@ -0,0 +1,2589 @@
+
+        return self
+
+    def get_email_destination_in_ccc(self):
+        """
+        Retrieve the details of the Email destination present in Cisco Catalyst Center.
+        Args:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+        Returns:
+            dict or None: A dictionary containing the details of the Email destination if it exists,
+                        otherwise returns None.
+        Description:
+            This function retrieves the details of the Email destination present in Cisco Catalyst Center.
+            If the Email destination exists, it returns a dictionary containing its details.
+            If no Email destination is found, it returns None.
+        """
+
+        try:
+            response = self.dnac._exec(
+                family="event_management",
+                function='get_email_destination'
+            )
+            self.log("Received API response from 'get_email_destination': {0}".format(str(response)), "DEBUG")
+
+            if not response:
+                self.log("There is no Email destination present in Cisco Catalyst Center", "INFO")
+                return response
+
+            return response[0]
+
+        except Exception as e:
+            self.status = "failed"
+            self.msg = "Error while getting the details of Email destination present in Cisco Catalyst Center: {0}".format(str(e))
+            self.log(self.msg, "ERROR")
+            self.check_return_status()
+
+    def collect_email_playbook_params(self, email_details):
+        """
+        Collects the parameters required for configuring Email destinations from the playbook.
+        Args:
+            self (object): An instance of a class used for interacting with Cisco Catalyst Center.
+            email_details (dict): A dictionary containing the Email destination details from the playbook.
+        Returns:
+            dict: A dictionary containing the collected parameters for configuring Email destinations.
+        Description:
+            This function collects the parameters required for configuring Email destinations from the playbook.
+            It extracts parameters such as 'fromEmail', 'toEmail', 'subject', and SMTP configurations
+            (primary and secondary) from the provided email_details dictionary.
+        """
+
+        playbook_params = {
+            'fromEmail': email_details.get('from_email'),
+            'toEmail': email_details.get('to_email'),
+            'subject': email_details.get('subject')
+        }
+
+        if email_details.get('primary_smtp_config'):
+            primary_smtp_details = email_details.get('primary_smtp_config')
+            playbook_params['primarySMTPConfig'] = {}
+            playbook_params['primarySMTPConfig']['hostName'] = primary_smtp_details.get('hostname')
+            playbook_params['primarySMTPConfig']['port'] = primary_smtp_details.get('port', "25")
```

Comment:
port 25 is default only if smtp_type is DEFAULT. 
port 587 is default if smtp_type is TLS.. 

Can you check and udpate... Get smpt_type first..

Reply:
added the port number to 25 if snmp_type is "DEFAULT", for other we have the option from 465 and 587 based on type(SSL, TLS)